### ISSUE TITLE
Dynamic width for panel

### ISSUE THREAD

**Issue by aleeraser** [home-sweet-gnome/dash-to-panel#1579]
It would be nice to have a dynamic width (or height, if placed vertically) for the panel, where all the elements are lined up and arranged in order to occupy as less space as possible.

This could also come with adjustable margin and/or padding between sections, similar of what is already present for icons.

Anyway, thanks for this amazing piece of software.

### Additional info

Currently, if the icons do not fit the fixed size panel, they're cropped like this:

![Screenshot-20220411110342-607x47](https://user-images.githubusercontent.com/10846607/162702953-7189fbbf-bf5d-47f5-952c-c6106ed281f0.png)

The request is for a dynamic width for the panel which re-size itself depending on the number of icons.

**Comment by jkuester** [comment]
Do not close! Dynamic width would be an awesome feature!

**Comment by AhmedElshiekh** [comment]
In fact, I think this feature would be much better, please pay attention and include it in the panel

**Comment by KierPrev** [comment]
I  second this

**Comment by jkuester** [comment]
Do not close! Dynamic width would be an awesome feature!

**Comment by jkuester** [comment]
Do not close! Dynamic width would be an awesome feature!

**Comment by charlesg99** [comment]
@jkuester I agree

**Comment by sanwablo** [comment]
Has this request ever been seriously considered?

**Comment by charlesg99** [comment]
An option to have the panel behave like a dock was added to the master branch. Also added a border-radius option to make it look nicer. Please try it out and report back, I'll add some panel padding/margin options soon to complement this. Thanks!

![Image](https://github.com/user-attachments/assets/6b1c9dba-618f-4b27-85ba-2d94c623db4d)

**Comment by xalt7x** [comment]
@charlesg99 ,
I've noticed conflict between "Dynamic panel height" and "Desktop button". When both are enabled, app icons and indicators distort (please see screenshot). Also, the highlight border of the "Desktop button" doesn't follow global border radius.

![Image](https://github.com/user-attachments/assets/7df857f1-eb85-48cb-b6fa-00ac455ad32c)

**Comment by charlesg99** [comment]
Thanks for testing, is that on a virtual machine? I have the show desktop displayed but and the icons are fine on my end. For the radius on the show desktop button hover, not much I can do as the css draws it, to make it look nicer you can increase its width so the gray left border is before the border radius of the panel starts, like this:

![Image](https://github.com/user-attachments/assets/8190a5e6-29da-4578-a86e-f13b00285620)

You can increase the show desktop button's width here:

![Image](https://github.com/user-attachments/assets/c9e90966-6ad9-4062-a8df-cad7b3f35644)

**Comment by xalt7x** [comment]
@charlesg99 

> Thanks for testing, is that on a virtual machine? I have the show desktop displayed and the icons are fine on my end.

Yes, it's Fedora 41 Silverblue (GNOME 47) installed on the virt-manager. Display resolution is 1024x768. Rendering improves if I increase resolution.

**Comment by charlesg99** [comment]
@xalt7x Yes, thank you I just saw the distorted icons and it indeed seems to be related to the show desktop button. It doesn't happen with every icon sizes so it's weird, but when it happens, removing the show desktop button fixes it. I'll look into it!

**Comment by charlesg99** [comment]
> I've noticed conflict between "Dynamic panel height" and "Desktop button". When both are enabled, app icons and indicators distort (please see screenshot). Also, the highlight border of the "Desktop button" doesn't follow global border radius.

Should now be fixed on master and panel padding options were added. Please let me know if you find any more issues.

**Comment by sanwablo** [comment]
@charlesg99 Let me know whether this is the place for this or whether I should start another thread.

I've noticed that the panel shows up from intellihide, even if the panel does not stretch all the way to the corners. For instance, I have it on the left and it shows up whenever I try to click on the "Go back one page" button on Firefox. Can this be "fixed" or is it not considered an issue?

**Comment by charlesg99** [comment]
@sanwablo Yes, not sure yet if I consider this a bug, the empty space is a surface where the mouse is tracked to keep the panel revealed, otherwise you would need to really aim at the panel as it wouldn't reveal unless you put your mouse at the edge of the monitor where the panel actually is. Unfortunately, as soon as that empty surface has the ability to track the mouse, by default it blocks all inputs. I see that Dash to Dock has the same behavior/problem, I have a few ideas on how to solve this, but not sure if I'll get much success. Anyway, this is now tracked in #2273.

**Comment by sanwablo** [comment]
@charlesg99 Thanks for your reply. For me, the problem is not so much that the panel shows up as that it prevents me from clicking on certain buttons of windows. So if it shows up when I want to go back a page on Firefox, Firefox does not receive the input. I'll follow the other issue from now on.
